### DISCUSSION TITLE
Using spack as package manager for your personal machine.

### DISCUSSION THREAD

**Discussion by vijaymocherla** [spack/spack#42935] [Q&A]
Does anyone use spack as a package manager on their daily driver or a personal workstation (that runs Linux)?I'm curious how would one configure such a setup? 

For users mostly working with opensource packages for scientific computing applications, I think spack can be a wonderful solution to keep their machines clean without breaking any dependencies. I've often had issues where building on particular package that only a particular compiler and a some simple update down stream breaks the entire build. Spack solves this problem, but too often I've had to move back to apt/yum or conda to build packages. 

Would like to know if anyone has tried using spack as go-to package mananger?

**Comment by teaguesterling** [comment]
I've been trying to move all my personal and work projects in this direction. I'm still not out of using the package manager for GUIs or things that are tightly coupled to the OS. I have used a combination of spack and language specific environment managers to keep requirements isolated and reproducible.

**Comment by alexmyczko** [comment]
When I tried to use it in Debian it failed to install visit, hdfview and everything I tried. I also got an official debian package, but fail to find testers. (ubuntu testers also welcome)

I definitely stick to native packages and if i have extra needs they can be applied to native src packages.

**Reply by vijaymocherla** [reply]
I have had some success in using it on ubuntu but, it isn't a complete replacement as I still use conda environments to sandbox some packages.

**Reply by alexmyczko** [reply]
did you try with this? https://sid.ethz.ch/debian/spack/2024/

**Reply by teaguesterling** [reply]
Two of the problems I seem to run into are related to working with is that language-specific packages and package managers. 

The package managers will install packages into an environment view and get deleted if the view is regenerated.

Additionally, modern packaging tools often define their language-specific dependencies natively, making for a mosaic installation of both spack and non-spack managed packages .

For that reason, I'll often create a spack env to handle requirements that are either expected to be provided by the OS or are stable/difficult/complex enough to be in Spack or need to be compiled. I'll then create a virtualenv, renv, etc. in the spack env and use the language specific package manager to install the dependencies from the pyproject/renv/requirements/etc. file. This helps ensure the package or project doesn't actually depend on spack to be reusable. 

It'd be great to create a little extension that supported reading python/R/rust/etc. requirements files as additional inputs to populate spack environments.
